integration-tests/src/test/java/com/redhat/service/bridge/integration/tests/steps/IngressSteps.java

```diff
@@ -64,16 +70,41 @@ public void sendCloudEventToIngressOfBridgeWithPathIsFailingWithHTTPResponseCode
         sendAndCheckCloudEvent(testBridgeName, cloudEvent, path, responseCode);
     }
 
-    @When("^send a cloud event to the Ingress of the Bridge \"([^\"]*)\" with path \"([^\"]*)\" and default headers is failing with HTTP response code (\\d+):$")
+    @When("^send a cloud event to the Ingress of the Bridge \"([^\"]*)\" with path \"([^\"]*)\" and headers (\"[^\"]+\":\"[^\"]+\"(?:,\"[^\"]+\":\"[^\"]+\")*) is failing with HTTP response code (\\d+):$")
     public void sendPlainCloudEventToIngressOfBridgeWithPathIsFailingWithHTTPResponseCode(String testBridgeName,
-            String path, int responseCode, String cloudEvent) {
-        sendAndCheckCloudEvent(testBridgeName, cloudEvent, path, getDefaultCloudEventHeaders(), responseCode);
+            String path, String headers, int responseCode, String cloudEvent) {
+        sendAndCheckCloudEventWithHeaders(testBridgeName, cloudEvent, path, parseHeaders(headers), responseCode);
     }
 
     private void sendAndCheckCloudEvent(String testBridgeName, String cloudEvent, String path, int responseCode) {
+        String testCloudEventId = getCloudEventId(cloudEvent);
+        String systemCloudEventId = UUID.randomUUID().toString();
+        context.getBridge(testBridgeName).newCloudEvent(testCloudEventId, systemCloudEventId);
```

Comment:
Could you factorize that part in a `storeCloudEventInContext` method ?

Reply:
What do you mean?

Reply:
sorry. factorize with https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/409/files/9b45aeb58faf2f3627631ab1e6be2ee7179fe2df#diff-2ad6b82b1a4cb7047164a210570ca973bca664d7b7628eedd7136efa944fd46bR94-R96

Reply:
I meant https://github.com/sutaakar/sandbox/blob/MGDOBR-385/integration-tests/src/test/java/com/redhat/service/bridge/integration/tests/steps/IngressSteps.java#L94-L96

Reply:
Yes, worth trying.